core/new-gui/src/app/common/service/user/user-dictionary/dictionary.service.ts

```diff
@@ -0,0 +1,518 @@
+        catchError(DictionaryService.handleErrorResponse),
+        shareReplay());
+
+    result.subscribe(); // causes post request to actually be sent (see cold observables)
+    return result;
+  }
+
+  /**
+   * Dictionary events are created every time a get/getAll/set/delete operation returns sucessfully
+   * @returns USER_DICT_EVENTs, see type definition
+   */
+  public dictionaryEventStream(): Observable<USER_DICT_EVENT> {
+    return this.dictionaryEventSubject.asObservable();
+  }
+
+  /**
+   * get a proxy dictionary: access a local copy of the backend dictionary as an object *synchronously*
+   * without callbacks or subscriptions.
+   * All instances of proxy dictionaries are synced (changing one affects the others).
+   * assignment and deletion of attributes cause requests to be sent to the backend
+   * and the proxy dictionary only changes if those requests are successful.
+   * *Warning: proxy dictionaries are synced w/ backend each getAll() and should be correct 99.99% of the time.
+   * However they may fall out of sync due to unforseen networking errors.*
+   * @returns Proxy of a local copy, Throws NotReadyError if initial getAll() request hasn't finished
+   */
+  public getUserDictionary(): UserDictionary {
+    if (!this.ready.value) { throw new NotReadyError('incomplete initialization of user-dictionary service'); }
+    return this.proxyUserDictionary();
+  }
+
+  /**
+   * get a proxy dictionary: access a local copy of the backend dictionary as an object *synchronously*
+   * without callbacks or subscriptions.
+   * All instances of proxy dictionaries are synced (changing one affects the others).
+   * assignment and deletion of attributes cause requests to be sent to the backend
+   * and the proxy dictionary only changes if those requests are successful.
+   * *Warning: proxy dictionaries are synced w/ backend each getAll() and should be correct 99.99% of the time.
+   * However they may fall out of sync due to unforeseen networking errors.*
+   * @returns Proxy of a local copy
+   */
+  public forceGetUserDictionary(): UserDictionary {
+    // gets user dictionary even if local dictionary isn't initialized
+    return this.proxyUserDictionary();
+  }
+
+  /**
+   * get a proxy dictionary: access a local copy of the backend dictionary as an object *synchronously*
+   * without callbacks or subscriptions.
+   * All instances of proxy dictionaries are synced (changing one affects the others).
+   * assignment and deletion of attributes cause requests to be sent to the backend
+   * and the proxy dictionary only changes if those requests are successful.
+   * *Warning: proxy dictionaries are synced w/ backend each getAll() and should be correct 99.99% of the time.
+   * However they may fall out of sync due to unforseen networking errors.*
+   * @returns Proxy of a local copy once initial getAll() is finished
+   */
+  public getUserDictionaryAsync(): Observable<UserDictionary> {
```

Comment:
the doc indicates this is "synchronous" however the method name indicates the opposite. is it a mistake?

Reply:
synchronous is supposed to describe the dictionary proxy object: it has synchronous reads and writes
I copied this description from the other getUserDictionary() functions, but I didn't take into account that this could be confusing